tests/test_pipeline_e2e.py

```
@@ -0,0 +1,304 @@
+            final=SALES_CORRECTIONS[0]["final"],
+            category="PROCESS", session=95,
+        )
+        assert result is not None
+        severity = result.get("outcome") or result.get("data", {}).get("severity")
+        assert severity in ("as-is", "minor", "moderate", "major", "discarded")
+
+    def test_graduation_across_sessions(self, fresh_brain):
+        for corr in SALES_CORRECTIONS[:3]:
+            _simulate_session(fresh_brain, corr)
+        lessons = fresh_brain._load_lessons()
+        process_lessons = [l for l in lessons if l.category == "PROCESS"]
```

Ruff (0.15.9)

[error] Ambiguous variable name: `l`

(E741)